linux/deploy_vm/deploy_vm_from_iso.yml

```diff
@@ -3,19 +3,57 @@
 ---
 # Deploy a new VM and install guest OS automatically from an ISO image
 #
+- name: "Set fact of the new VM default hardware version"
+  ansible.builtin.set_fact:
+    hardware_version: "{{ esxi_latest_hardware_version }}"
+  when: hardware_version is undefined
+
+- name: "Get '{{ guest_id }}' guest config options with VM hardware version '{{ hardware_version }}'"
+  include_tasks: ../../common/esxi_get_guest_config_options.yml
+  vars:
+    esxi_hardware_version: "{{ hardware_version }}"
+
 # Initialize undefined variables
 - name: "Initialize variables for new VM settings"
   ansible.builtin.set_fact:
-    boot_disk_controller: "{{ boot_disk_controller | lower if (boot_disk_controller is defined and boot_disk_controller) else 'paravirtual' }}"
-    network_adapter_type: "{{ network_adapter_type | lower if (network_adapter_type is defined and network_adapter_type) else 'vmxnet3' }}"
-    firmware: "{{ firmware | lower if (firmware is defined and firmware) else 'efi' }}"
+    boot_disk_controller: >-
+      {{ 
+         (boot_disk_controller is defined and boot_disk_controller) 
+         | ternary(boot_disk_controller | lower, guest_config_options.default_disk_controller) 
+      }}
+    network_adapter_type: >-
+      {{
+         (network_adapter_type is defined and network_adapter_type) 
+         | ternary(network_adapter_type | lower, guest_config_options.default_ethernet) 
+      }}
+    firmware: >-
+      {{
+         (firmware is defined and firmware) 
+         | ternary(firmware | lower, guest_config_options.rec_firmware) 
+      }} 
     secureboot_enabled: "{{ secureboot_enabled is defined and secureboot_enabled | bool }}"
```

Comment:
If `secureboot_enabled is not set`, shall here use `guest_config_options.default_secure_boot`?